Documentation/specs/corehost.md

```diff
@@ -0,0 +1,112 @@
+# corehost runtime/assembly resolution
+
+The shared host locates assemblies and native libraries using a combination of: Servicing Index, Files in the application folder (aka "app-local") and files from package caches.
+
+## Definitions and Formats
+
+### Servicing Index
+
+The servicing index is loaded when the `DOTNET_SERVICING` environment variable is non-empty. When this variable is non-empty, it points to a directory that will be the **Servicing Root**.
+
+An index file is located at the path defined by `$DOTNET_SERVICING/servicing_index.txt`. In this file are a series of lines of one of the following formats:
```

Comment:
A couple things:

1. I don't have full context on the history of this, so @davidfowl or @moozzyk might have better context (this came from DNX)

2. Servicing is not expected to occur very often, it is for critical security issues only. This file shouldn't get too big.

Having said that, perhaps a two-level index (first level indicating packages that have been serviced by ID, second level providing Versions and Paths serviced) might provide a better design. I'll sync with the folks on DNX who designed the original plan to see what they think.